tests/test_cli.py

```diff
@@ -72,5 +72,5 @@ def _capture_match_gpus_spec(file_path, gpus: str):
 
         _capture_match_gpus_spec(f.name, 'V100:1')
         _capture_match_gpus_spec(f.name, 'v100:1')
-        _capture_match_gpus_spec(f.name, 'V100:0.5')
         _capture_match_gpus_spec(f.name, 'V100')
+        _capture_match_gpus_spec(f.name, 'V100:0.5')
```

Comment:
I feel it's a bit weird that partial GPUs are only supported when the instance type is specified, while they are not supported otherwise. Why don't we disallow partial GPUs for `sky launch`?

```
$ sky launch --gpus v100:0.5
I 05-11 13:23:15 optimizer.py:998] No resource satisfying <Cloud>({'V100': 0.5}) on [AWS, Azure, GCP, Lambda].
I 05-11 13:23:15 optimizer.py:1001] Did you mean: ['V100-32GB:8', 'V100:1', 'V100:2', 'V100:4', 'V100:8']
sky.exceptions.ResourcesUnavailableError: No launchable resource found for task Task(run=<empty>)
  resources: <Cloud>({'V100': 0.5}).
This means the catalog does not contain any instance types that satisfy this request.
To fix: relax or change the resource requirements.
Hint: 'sky show-gpus --all' to list available accelerators.
      'sky check' to check the enabled clouds.

$ sky launch --gpus v100:0.5 --instance-type p3.2xlarge
I 05-11 13:23:22 optimizer.py:636] == Optimizer ==
I 05-11 13:23:22 optimizer.py:647] Target: minimizing cost
I 05-11 13:23:22 optimizer.py:659] Estimated cost: $3.1 / hour
I 05-11 13:23:22 optimizer.py:659] 
I 05-11 13:23:22 optimizer.py:732] Considered resources (1 node):
I 05-11 13:23:22 optimizer.py:781] -----------------------------------------------------------------------------------------
I 05-11 13:23:22 optimizer.py:781]  CLOUD   INSTANCE     vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN   
I 05-11 13:23:22 optimizer.py:781] -----------------------------------------------------------------------------------------
I 05-11 13:23:22 optimizer.py:781]  AWS     p3.2xlarge   8       61        V100:0.5       us-east-1     3.06          ✔     
I 05-11 13:23:22 optimizer.py:781] -----------------------------------------------------------------------------------------
I 05-11 13:23:22 optimizer.py:781] 
Launching a new cluster 'sky-e1f9-woosuk'. Proceed? [Y/n]:
```